features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/ByteBufferXmlAdapter.java

```diff
@@ -28,13 +28,20 @@
 
 package org.opennms.netmgt.syslogd;
 
-import org.junit.Test;
+import java.nio.ByteBuffer;
 
-public class SyslogDTOMapperTest {
+import javax.xml.bind.annotation.adapters.XmlAdapter;
 
-	@Test
-	public void test() {
-		// TODO: Implement test
-	}
+public class ByteBufferXmlAdapter extends XmlAdapter<byte[], ByteBuffer> {
+
+    @Override
+    public ByteBuffer unmarshal(byte[] bytes) throws Exception {
+        return ByteBuffer.wrap(bytes);
+    }
+
+    @Override
+    public byte[] marshal(ByteBuffer bb) throws Exception {
+        return bb.array();
```

Comment:
This call relies on optional behavior of ByteBuffer. It's safer to use the original code:
```
    @XmlAttribute
    public byte[] getBytes() {
        m_bytes.rewind();
        byte[] retval = new byte[m_bytes.remaining()];
        m_bytes.get(retval);
        m_bytes.rewind();
        return retval;
    }
```
If you want to preserve the position of the buffer, you'll have to add code for that (but it's probably not a problem here).